bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/obscomp/SPSiteQuality.java

```diff
@@ -666,24 +673,30 @@ public void setElevationConstraintMax(double max) {
         return Collections.unmodifiableList(_timingWindows);
     }
 
-    public void setTimingWindows(List<TimingWindow> windows) {
-        List<TimingWindow> prev = Collections.unmodifiableList(new ArrayList<>(_timingWindows));
-        _timingWindows.clear();
-        _timingWindows.addAll(windows);
-        firePropertyChange(TIMING_WINDOWS_PROP.getName(), prev, getTimingWindows());
+    // Common method to make changes to the timing window list and then fire a property change if appropriate.
+    private void changeTimingWindows(final Supplier<Boolean> changer) {
```

Comment:
There was quite a bit of repeated code in the timing window modifiers, so I abstracted this out to minimize and simplify.